### PR TITLE
Support strict Text

### LINKED ISSUE #6: Pretty instance for strict Text
Is there a reason that strict Text is not part of the API at all? It would be convenient to have variants for strict Text and an instance of `Pretty`. Even better if it were polymorphic over both when possible.

### PR DESCRIPTION
**Pull request by 3noch**
Fixes #6

### CODE REVIEW

File: Text/PrettyPrint/Leijen/Text.hs [hunk 1]
```diff
@@ -126,7 +126,9 @@ import Data.String (IsString (..))
 import System.IO   (Handle, hPutChar, stdout)
 
 import           Data.Int               (Int64)
+import           Data.List              (intersperse)
 import           Data.Monoid            (Monoid (..), (<>))
+import qualified Data.Text              as Ts
```
Comment (ivan-m) [inline Text/PrettyPrint/Leijen/Text.hs]: Can the import be `ST` or `TS` please? Having a lowercase there looks a bit odd.

File: Text/PrettyPrint/Leijen/Text.hs [hunk 2]
```diff
@@ -516,6 +518,10 @@ string str = case T.uncons str of
                _                -> case (T.span (/='\n') str) of
                                      (xs,ys) -> text xs <> string ys
 
+stringStrict :: Ts.Text -> Doc
+stringStrict = mconcat . intersperse line . map textStrict . Ts.lines
+
```
Comment (ivan-m) [inline Text/PrettyPrint/Leijen/Text.hs]: I can't believe I didn't do something like this for the normal `string` case... would you mind fixing that up as well please?

**Comment by ivan-m** [comment]
(Ignore the "check" failures; that's because Travis decided to turn itself on without a .travis.yml file to say what to do.)

**Comment by ivan-m** [comment]
To fully support strict `Text`, we'd also need a strict variant of `displayT` (`displayST` maybe?).

Would you also be able to make these changes to the `Monadic` module please?

**Comment by 3noch** [comment]
I'll add those today.

**Comment by 3noch** [comment]
I named it `displayTStrict` for consistency but I'm happy to name it whatever you want.

File: Text/PrettyPrint/Leijen/Text.hs [hunk 3]
```diff
@@ -513,6 +514,10 @@ equals = char '='
 string :: Text -> Doc
 string = mconcat . intersperse line . map text . T.lines
 
+stringStrict :: TS.Text -> Doc
+stringStrict = mconcat . intersperse line . map textStrict . TS.lines
+
+
```
Comment (ivan-m) [inline Text/PrettyPrint/Leijen/Text.hs]: One too many newlines here.

File: Text/PrettyPrint/Leijen/Text.hs [hunk 4]
```diff
@@ -1005,6 +1018,9 @@ c `consB` b = B.singleton c `mappend` b
 displayT :: SimpleDoc -> Text
 displayT = B.toLazyText . displayB
 
+displayTStrict :: SimpleDoc -> TS.Text
+displayTStrict = T.toStrict . displayT
```
Comment (3noch) [inline Text/PrettyPrint/Leijen/Text.hs]: Hmm, I wasn't able to find one. Where is it? I think there might be one in `text-show` which would be a new dependency.

Comment (ivan-m) [inline Text/PrettyPrint/Leijen/Text.hs]: Nevermind, I must have gotten it mixed up with `Bytestring`'s Builder.

File: Text/PrettyPrint/Leijen/Text.hs (hunk 3 above)
Comment (3noch) [inline Text/PrettyPrint/Leijen/Text.hs]: Fixed.

**Comment by 3noch** [comment]
@ivan-m Fixed whitespace.

**Comment by ivan-m** [comment]
Same thing with `stringStrict` and `textStrict` in `Monadic`.

**Comment by 3noch** [comment]
Didn't I fix it there too? I thought it did.

**Comment by ivan-m** [comment]
OK, GitHub's diff lied to me earlier.  Thanks!

**Comment by 3noch** [comment]
Thank you!

**Pull request merged**